ServerCore/Pages/Threads/PuzzleThreads.cshtml.cs

```diff
@@ -149,20 +149,34 @@ public PuzzleThreadsModel(PuzzleServerContext serverContext, UserManager<Identit
                 throw new NotSupportedException($"EventRole [{EventRole}] is not supported in PuzzleThreads");
             }
 
-            // Filter down to only latest messages
-            LatestMessagesFromEachThread = await messages
-                .GroupBy(message => message.ThreadId)
-                .Select(group => group.OrderByDescending(message => message.CreatedDateTimeInUtc).First())
-                .ToListAsync();
+            // Filter down to only latest messages in memory
+            List<Message> retrievedMessages = await messages.ToListAsync();
```

Comment:
If you do a Select to an object with a subset of Message's properties (only the ones you use), you can have far less traffic between the DB and the front end. The biggie to skip is Message.Text, which could be large. See MemberView in Players.cshtml.cs for an example.

Reply:
I have the text to show the preview of the most recent message in the output. 
I'm fine to remove it, but just wondering if it was helpful at all to have the message preview

Reply:
I thought it was title only - my apologies. But sending all the message text makes me concerned that this won't be a perf win in the long run. Let's wait to see if @morganbr has other ideas...